internal/pkg/cli/app_init.go

```diff
@@ -0,0 +1,231 @@
+func (opts *AppInitOpts) Execute() error {
+	manifest, err := manifest.Create(opts.AppName, opts.AppType, opts.DockerfilePath)
+	if err != nil {
+		return fmt.Errorf("failed to generate a manifest %w", err)
+	}
+	manifestBytes, err := manifest.Marshal()
+	if err != nil {
+		return fmt.Errorf("failed to marshal the manifest file %w", err)
+	}
+	manifestPath, err := opts.manifestWriter.WriteManifest(manifestBytes, opts.AppName)
+	if err != nil {
+		return fmt.Errorf("failed to write manifest for app %s: %w", opts.AppName, err)
+	}
+
+	wkdir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+	relPath, err := filepath.Rel(wkdir, manifestPath)
+	if err != nil {
+		return fmt.Errorf("failed to get relative path of manifest file: %w", err)
+	}
+	log.Infoln()
+	log.Successf("Wrote the manifest for %s app at '%s'\n", color.HighlightUserInput(opts.AppName), color.HighlightResource(relPath))
+	log.Infoln()
+	return nil
+}
+
+func (opts *AppInitOpts) askAppType() error {
+	t, err := opts.prompt.SelectOne(
+		"What type of application do you want to make?",
+		"List of infrastructure patterns.",
+		manifest.AppTypes)
+
+	if err != nil {
+		return fmt.Errorf("failed to get type selection: %w", err)
+	}
+	opts.AppType = t
+	return nil
+}
+
+func (opts *AppInitOpts) askAppName() error {
+	name, err := opts.prompt.Get(
+		fmt.Sprintf("What do you want to call this %s?", opts.AppType),
+		"Collection of AWS services to achieve a business capability. Must be unique within a project.",
+		validateApplicationName)
+	if err != nil {
+		return fmt.Errorf("failed to get application name: %w", err)
+	}
+	opts.AppName = name
+	return nil
+}
+
+// askDockerfile prompts for the Dockerfile by looking at sub-directories with a Dockerfile.
+// If the user chooses to enter a custom path, then we prompt them for the path.
+func (opts *AppInitOpts) askDockerfile() error {
+	dockerfiles, err := opts.listDockerfiles()
+	if err != nil {
+		return err
+	}
+	const customPathOpt = "Enter a custom path"
+	selections := make([]string, len(dockerfiles))
+	copy(selections, dockerfiles)
+	selections = append(selections, customPathOpt)
+
+	sel, err := opts.prompt.SelectOne(
+		fmt.Sprintf("Which Dockerfile would you like to use for %s app?", opts.AppName),
+		"Dockerfile to use for building your application's container image.",
+		selections,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to select Dockerfile: %w", err)
+	}
+
+	if sel == customPathOpt {
+		sel, err = opts.prompt.Get("OK, what's the path to your Dockerfile?", "", nil)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to get Dockerfile: %w", err)
+	}
+	opts.DockerfilePath = sel
+	return nil
+}
+
+// listDockerfiles returns the list of Dockerfiles within a sub-directory below current working directory.
+// If an error occurs while reading directories, returns the error.
+func (opts *AppInitOpts) listDockerfiles() ([]string, error) {
```

Comment:
I feel like we should move this method to workspace and call it
`listDockerfiles` > it returns a LocalDockerfile struct:

```golang

type LocalDockerfile struct {
 path string
 linkedApp string
 linkedToApp bool
}
```

That way we can only let folks select apps that haven't been selected yet. We can also print a debug line saying that some were excluded because they already belonged to a particular app.

(does that make sense)? Feel free to put a TODO to address this and come back to it in a different PR